sklearn/cluster/_spectral.py

```diff
@@ -195,10 +195,14 @@ def spectral_clustering(affinity, n_clusters=8, n_components=None,
         but may also lead to instabilities
 
     random_state : int, RandomState instance, default=None
-        A pseudo random number generator used for the initialization of the
-        lobpcg eigen vectors decomposition when eigen_solver == 'amg' and by
-        the K-Means initialization. Use an int to make the randomness
-        deterministic.
+        A pseudo random number generator used for the initialization
+        of the lobpcg eigen vectors decomposition when eigen_solver ==
+        'amg' and by the K-Means initialization. Use an int to make
+        the randomness deterministic. When using ``solver`` == 'amg'
+        it is necessary to also fix the global numpy seed with
+        ``np.random.seed(int)`` to get deterministic results. See
+        https://github.com/pyamg/pyamg/issues/139 for further
+        information.
         See :term:`Glossary <random_state>`.
```

Comment:
A few minor suggested fixes:

- solver -> eigen_solver
- by -> for
- move up glossary link to avoid potential confusion
- make a note about the pyamg-specific issue

Would you mind applying similar fixes to the other entries @jnboehm ? Thanks!

```suggestion
        A pseudo random number generator used for the initialization
        of the lobpcg eigen vectors decomposition when `eigen_solver ==
        'amg'`, and for the K-Means initialization. Use an int to make
        the get deterministic results across calls (See
        :term:`Glossary <random_state>`).
        
        .. note::
        	When using `eigen_solver` == 'amg'`,
        	it is necessary to also fix the global numpy seed with
        	`np.random.seed(int)` to get deterministic results. See
        	https://github.com/pyamg/pyamg/issues/139 for further
        	information.

```

Reply:
Thank you for the suggestions, it works better this way.  I've applied the changes to all four relevant doc strings.